### PR TITLE
tools/crimson: crimson tool improvement

### PR DESCRIPTION
**Pull request by aisakaki**
1. support task log, will save all tasks log to target directory
2. save OSD logs for each tester
3. collect multicore reactor utilization
4. introduce Time Continuous Tasks, this will help when there is multiple time points task, for example ,to collect ru at three time point, this mechanism will automatically calculate the right time points to run the task.
5. support multicore list test
6. some codes cleanup

### CODE REVIEW

File: tools/crimson/README.rst [hunk 1]
```diff
@@ -95,27 +94,28 @@ ceph cluster because the tool will do that. Put this tool in the ceph build
...
 
 .. code-block:: console
-    
-    sudo ./crimson_stress_tool.py \ 
-        --client-list 4 8 --thread-list 2 4 6 --bench-taskset 16-31 --time 60 \
-        --block-size 4K \
+
+    sudo ./crimson_bench_tool.py \
+        --client 4 8 --thread 2 4 6 --bench-taskset 16-31 --time 60 \
+        --block-size 4096
```
Comment (cyx1231st) [inline tools/crimson/README.rst:105]: missing \

File: tools/crimson/crimson_bench_tool.py [hunk 2]
```diff
@@ -1031,26 +1112,37 @@ def rados_pre_write(self, block_size, thread_num, time):
                         help='ratio of fio seq read clients')
 
     # time point based thread param
-    parser.add_argument('--reactor-utilization',
+    parser.add_argument('--ru',
+                        type=int,
+                        help='how many time point to collect the \
+                            reactor utilization')
+    parser.add_argument('--iostat',
+                        action='store_true',
```
Comment (cyx1231st) [inline tools/crimson/crimson_bench_tool.py:1120]: should be type=int ?

File: tools/crimson/crimson_bench_tool.py [hunk 3]
```diff
@@ -1031,26 +1112,37 @@ def rados_pre_write(self, block_size, thread_num, time):
...
+                        help='how many time point to collect iostat information')
+    parser.add_argument('--freq',
                         action='store_true',
```
Comment (cyx1231st) [inline tools/crimson/crimson_bench_tool.py:1123]: should be type=int ?

File: tools/crimson/README.rst [hunk 4]
```diff
@@ -95,27 +94,28 @@ ceph cluster because the tool will do that. Put this tool in the ceph build
...
-        --freq
+        --freq 6
 
 The tool will run rados bench write and read test case with the combination 
 of 4 or 8 clients and 2, 4 or 6 threads. In Every test case, there will be 75% of
 write clients in all clients and the read clients will be 25%. Also, you can set
 read clients ratio to 0 to do the write only tests, vive versa.
-Meanwhile, it will collect the reactor cpu utilization, and the perf information. 
+Meanwhile, it will collect the reactor cpu utilization for three time points(15,
+30, 45) for this case, and the perf information.
```
Comment (cyx1231st) [inline tools/crimson/README.rst:118]: and also get the cpu frequence 6 times

File: tools/crimson/crimson_bench_tool.py [hunk 5]
```diff
@@ -991,10 +1072,10 @@ def rados_pre_write(self, block_size, thread_num, time):
...
                         default = 1,
                         help='how many osds')
-    parser.add_argument('--smp',
-                        type=int,
-                        default = 0,
-                        help='core per osd')
+    parser.add_argument('--log',
+                        type=str,
+                        default = None,
+                        help='directory to store logs, no log by default')
```
Comment (cyx1231st) [inline tools/crimson/crimson_bench_tool.py:1078]: Please summarize which information will be stored in the directory.

File: tools/crimson/crimson_stress_tool.py [hunk 6]
```diff
@@ -486,10 +471,10 @@ def post_process(self, test_case_result):
             print("cannot find flamegraph scripts, will not generate flamegraph.")
         return
 
+
```
Comment (cyx1231st) [inline tools/crimson/crimson_stress_tool.py:474]: unrelated change

**Review by cyx1231st — COMMENTED** [review]
Please correct the titles as "tools/crimson:"

File: tools/crimson/crimson_stress_tool.py (hunk 6 above)
Comment (aisakaki) [reply tools/crimson/crimson_stress_tool.py]: all issues above addressed~

**Review by cyx1231st — APPROVED** [review]
